rules/DeadCode/Rector/ClassMethod/RemoveUnusedPublicMethodParameterRector.php

```diff
@@ -75,7 +76,11 @@ public function getNodeTypes(): array
     public function refactor(Node $node): ?Node
     {
         // may have child, or override parent that needs to follow the signature
-        if (! $node->isFinal() || $node->extends instanceof FullyQualified || $node->implements !== []) {
+        if (! $node->isFinal() && FeatureFlags::treatClassesAsFinal() === false) {
```

Comment:
it cause buggy on abstract method to be implemented by child:

```php
abstract public function run($a, $b);
```

since `final` class method won't have `abstract` method, but `non-final` class can have, so this needs check 

```php
if ($classMethod->isAbstract()) {
   return true;
}
```

in `shouldSkipClassMethod()`

Reply:
sure, I created PR:

- https://github.com/rectorphp/rector-src/pull/7002

for it.